src/models/StyledComponent.js

```diff
@@ -121,43 +121,45 @@ export default (ComponentStyle: Function, constructWithOptions: Function) => {
 
     componentWillMount() {
       const { componentStyle } = this.constructor
-      const styledContext = this.context[CHANNEL_NEXT]
+      if (this.context) {
```

Comment:
This will have to be more explicit so v8 doesn’t deopt this check. “!== undefined” will do